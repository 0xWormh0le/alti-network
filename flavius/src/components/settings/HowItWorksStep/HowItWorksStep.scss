@import 'styles/_shared.scss';

.HowItWorksStep {
  &__title {
    color: $TEXT_COLOR;
    margin-bottom: 1rem;
  }

  &__icon {
    @include icon();
  }
}
