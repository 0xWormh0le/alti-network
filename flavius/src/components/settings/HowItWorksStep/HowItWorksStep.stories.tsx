import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import HowItWorksStep, { HowItWorksStepProps } from './HowItWorksStep'

export default {
  title: 'Settings/HowItWorksStep',
  component: HowItWorksStep
} as Meta

const Template: Story<HowItWorksStepProps> = (args) => <HowItWorksStep {...args} />

export const Default = Template.bind({})
Default.args = {}
