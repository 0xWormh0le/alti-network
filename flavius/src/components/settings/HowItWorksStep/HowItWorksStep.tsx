import React from 'react'
import Button from 'components/elements/Button'
import Typography, { TypographyVariant } from 'components/elements/Typography'
import padlock from 'icons/padlock.svg'
import smartPhone from 'icons/smartphone.svg'
import verified from 'icons/verified.svg'
import UI_STRINGS from 'util/ui-strings'
import './HowItWorksStep.scss'

export interface HowItWorksStepProps {
  onStepCompleted: () => void
}

const HowItWorksStep: React.FC<HowItWorksStepProps> = ({ onStepCompleted }) => (
  <section className='HowItWorksStep TwoFactorAuthentication__step'>
    <Typography variant={TypographyVariant.H3} weight='semibold' className='HowItWorksStep__title'>
      {UI_STRINGS.SETTINGS.HOW_TWO_FACTOR_AUTH_WORKS}
    </Typography>
    <Typography variant={TypographyVariant.BODY_LARGE}>
      <img src={padlock} className='HowItWorksStep__icon' alt='padlock' />
      {UI_STRINGS.SETTINGS.WHEN_LOGGING_IN_YOU_WILL_NEED}
    </Typography>
    <Typography variant={TypographyVariant.BODY_LARGE}>
      <img src={smartPhone} className='HowItWorksStep__icon' alt='smartphone' />
      {UI_STRINGS.SETTINGS.LOGIC_CODE_WILL_BE_SENT}
    </Typography>
    <Typography variant={TypographyVariant.BODY_LARGE}>
      <img src={verified} className='HowItWorksStep__icon' alt='checkmark' />
      {UI_STRINGS.SETTINGS.WHEN_YOU_TYPE_YOU_WILL_KNOW}
    </Typography>
    <Button
      className='HowItWorksStep__button'
      action='primary'
      text={UI_STRINGS.BUTTON_LABELS.START}
      onClick={onStepCompleted}
    />
  </section>
)

export default HowItWorksStep
