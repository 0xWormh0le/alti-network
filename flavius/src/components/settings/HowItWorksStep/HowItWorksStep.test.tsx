import React from 'react'
import { render } from '@testing-library/react'

import noop from 'lodash/noop'
import HowItWorksStep from './HowItWorksStep'

describe('HowItWorksStep', () => {
  it('renders correctly', () => {
    const localProps = {
      onStepCompleted: noop
    }
    const { container } = render(<HowItWorksStep {...localProps} />)
    expect(container).toMatchSnapshot()
  })
})
