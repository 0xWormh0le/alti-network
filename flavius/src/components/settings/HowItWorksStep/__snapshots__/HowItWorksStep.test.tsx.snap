// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowItWorksStep renders correctly 1`] = `
<div>
  <section
    class="HowItWorksStep TwoFactorAuthentication__step"
  >
    <h3
      class="Typography Typography--h3 Typography--semibold HowItWorksStep__title"
    >
      How Two-Factor Authentication works
    </h3>
    <p
      class="Typography Typography--body-large"
    >
      <img
        alt="padlock"
        class="HowItWorksStep__icon"
        src="padlock.svg"
      />
      When logging in, you'll need to provide a login code.
    </p>
    <p
      class="Typography Typography--body-large"
    >
      <img
        alt="smartphone"
        class="HowItWorksStep__icon"
        src="smartphone.svg"
      />
      The logic code will be sent to you via a mobile app, which uses TOTP protocol.
    </p>
    <p
      class="Typography Typography--body-large"
    >
      <img
        alt="checkmark"
        class="HowItWorksStep__icon"
        src="verified.svg"
      />
      When you type in the login code, we'll know it's really you.
    </p>
    <button
      class="Button HowItWorksStep__button Button--primary"
      type="button"
    >
      Start
    </button>
  </section>
</div>
`;
